### PR TITLE
Toggle switch 0

### PR DESCRIPTION
**Pull request by lfbergee**
## 📥 Proposed changes

Toggle switch style and react component

## ☑️ Submission checklist

-   [x] I have read the [CONTRIBUTING](https://github.com/fremtind/jokul/blob/master/CONTRIBUTING.md) doc
-   [x] `yarn build` works locally with my changes
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] I have added necessary documentation (if appropriate)

## 💬 Further comments

This implementation is not as elegant as the checkbox, however this will work for older browsers as well.

CLOSES: #19

### CODE REVIEW

File: packages/toggle-switch/toggle-switch.scss [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+$toggle-focus-outline-color: $himmel;
+
+.jkl-toggle-switch {
+    & input {
+        opacity: 0;
+        width: 0;
+        height: 0;
+    }
+    position: relative;
+    min-width: $toggle-width * 5;
```
Comment (piofinn) [inline packages/toggle-switch/toggle-switch.scss]: This should not be needed. I have a small refactor that gets rid of this and cleans up a few other things

File: packages/toggle-switch/toggle-switch.scss [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    & input:checked + .jkl-toggle-switch__slider:after {
+        transform: translateX(rem(30px)) scale(1.2);
+    }
+    html:not([data-mousenavigation]) & input:focus {
+        + .jkl-toggle-switch__slider {
+            box-shadow: 0 0 0 rem(2px) $snøhvit, 0 0 0 rem(4px) $toggle-focus-outline-color;
+        }
+    }
+    &__label {
+        position: absolute;
```
Comment (piofinn) [inline packages/toggle-switch/toggle-switch.scss]: No need to position this absolutely. Again, I have a fix for this in the refactor

**Comment by lfbergee** [comment]
Yeah, I think i got a bit confused with different sketches, so yeah, i wasn't sure about the states. So did the most complicated thing variant, so we can simplify.

File: packages/toggle-switch/toggle-switch.scss (hunk 1 above)
Comment (lfbergee) [reply packages/toggle-switch/toggle-switch.scss]: Cool, yeah, this should be aviodable

File: packages/toggle-switch-react/src/ToggleSwitch.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React, { ReactNode } from "react";
+
+interface Props {
```
Comment (piofinn) [inline packages/toggle-switch-react/src/ToggleSwitch.tsx]: Shouldn't there be an onChange prop?

Comment (lfbergee) [reply packages/toggle-switch-react/src/ToggleSwitch.tsx]: Probably, I work out from checkbox, and it does not have it. But yeah, it should be easy to hook into from a parent component

Comment (piofinn) [reply packages/toggle-switch-react/src/ToggleSwitch.tsx]: Huh, probably just a slip-up in checkbox as well. In my mind both will be used as controlled inputs more often than not

File: packages/toggle-switch-react/src/ToggleSwitch.tsx [hunk 4]
```diff
@@ -4,12 +4,18 @@ interface Props {
...
-export const ToggleSwitch = ({ children, defaultChecked, className = "" }: Props) => {
+export const ToggleSwitch = ({ children, defaultChecked, onChange, className = "" }: Props) => {
     return (
         <label data-testid="jkl-toggle-input--label" className={`jkl-toggle-switch ${className}`}>
-            <input data-testid="jkl-toggle-input" type="checkbox" defaultChecked={defaultChecked} />
+            <input
+                data-testid="jkl-toggle-input"
+                type="checkbox"
+                defaultChecked={defaultChecked}
+                onChange={(e) => onChange && onChange(e.target.checked)}
```
Comment (lfbergee) [inline packages/toggle-switch-react/src/ToggleSwitch.tsx]: Should be a named function

File: packages/toggle-switch-react/src/ToggleSwitch.test.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import React from "react";
```
Comment (lfbergee) [inline packages/toggle-switch-react/src/ToggleSwitch.test.tsx]: We should add a test for the onchange handler

Comment (piofinn) [reply packages/toggle-switch-react/src/ToggleSwitch.test.tsx]: Done :)

**Pull request merged**